Assets/_BForBoss/_Core/Scripts/HUD/CountdownViewBehaviour.cs

```diff
@@ -12,19 +12,36 @@ public class CountdownViewBehaviour : MonoBehaviour
         private bool _isRunning;
 
         private float _amountOfTime;
+        
+        private Action OnCountdownCompleted;
+        
+        public bool IsRunning => _isRunning;
 
-        public void StartCountdown(float amountOfTime)
+        public void StartCountdown(float amountOfTime, Action onCountdownCompleted = null)
```

Comment:
Imo, the count down HUD / UI, shouldn't be the one that determines it ended. 

The logic of "The time has ended, now do this" should be contained within the Manager itself. 
This UI/HUD behaviour should soley be in charge of letting users know the time.

Reply:
Yeah that was mostly a result of how CountdownViewBehaviour currently is setup, I can rewrite it to be more of an actual view though

Reply:
I just remembered countDownViewBehaviour is more than not a placeholder. This part is flexible